projects/angular/src/forms/common/if-control-state/if-error.ts

```diff
@@ -35,27 +35,30 @@ export class ClrIfError extends AbstractIfState {
    * @param state CONTROL_STATE
    */
   protected override handleState(state: CONTROL_STATE) {
-    if (this.error && this.control) {
+    if (this.error && this.control && !!this.control.invalid) {
       this.displayError(this.control.hasError(this.error));
+    } else if (this.error && !!this.additionalControls?.length) {
+      const invalidControl = this.additionalControls?.filter(control => control.hasError(this.error))[0];
```

Comment:
Is it impossible that the filter will return 0 elements and trying to access [0] will throw an error?
Or we rely on the assumption that some control set this.error and this is not the main control, but _should_ be some of the additionalControls?

Reply:
array[0] will return undefined only, It wont be returning any error. Even in `strict mode` the behavior is same. Its already gracefully handled in subsequent methods (displayError()) where this undefined value is passed.